assets/js/googlesitekit/modules/datastore/sharing-settings.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * `core/modules` data store: module sharing settings
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { createFetchStore } from '../../data/create-fetch-store';
+import { CORE_MODULES } from './constants';
+
+// Actions
+const SET_SHARING_MANAGEMENT = 'SET_SHARING_MANAGEMENT';
+const SET_SHARED_ROLES = 'SET_SHARED_ROLES';
+const RECEIVE_GET_SHARING_SETTINGS = 'RECEIVE_GET_SHARING_SETTINGS';
+
+const baseInitialState = {
+	sharingSettings: undefined,
+	savedSharingSettings: undefined,
+};
+
+const fetchSaveSharingSettingsStore = createFetchStore( {
+	baseName: 'saveSharingSettings',
+	controlCallback: ( { savedSharingSettings } ) => {
+		return API.set(
+			'core',
+			'modules',
+			'sharing-settings',
+			{ ...savedSharingSettings },
```

Comment:
No need to destruct the `savedSharingSettings` object, just pass it as is:
```suggestion
			savedSharingSettings,
```